components/server/src/ide-config.ts

```diff
@@ -190,6 +191,23 @@ export class IDEConfigService {
                 }
             }
 
+            for (const [id, option] of Object.entries(newValue.ideOptions.options).filter(([_, x]) => x.latestImage)) {
```

Comment:
this for loop is fairly complex, with many things happening in a single line - did you consider breaking it down a bit?